### ISSUE TITLE
The linter finds incorrect diagnostics when there is an inline comment

### ISSUE THREAD

**Issue by Sp0k** [dotenv-linter/dotenv-linter#808]
When using the tool on my `.env` file, the tool gets confused with inline comments and recognizes them as values without quotes:
```bash
DATABASE_HOST=test           # This comment will create an issue
DB_PASSWORD=password         # This comment wil not for some reason
RAILS_HOSTS=host             # This comment will create an issue
```
 
When I run the `.env` file through the linter, this is the output:
```bash
.env:1 ValueWithoutQuotes: This value needs to be surrounded in quotes
.env:2 ValueWithoutQuotes: This value needs to be surrounded in quotes
.env:3 ValueWithoutQuotes: This value needs to be surrounded in quotes
```

But when I change the file to add quotes on the comment it stops:

```bash
DATABASE_HOST="test"          # "This comment will no longer create an issue"
DB_PASSWORD="password"        # "This comment will no longer create an issue"
RAILS_HOSTS="host"            # "This comment will no longer create an issue"
```

Output:
```bash
No problems found
```
